### PR TITLE
[iOS][expo-video] Add custom DRM license request option

### PR DESCRIPTION
**Pull request by JustJoostNL**
# Why

The new `expo-video` library should support a custom way to get the DRM license.

# How

I added `getLicense` to the `DRMOptions`. This function is called (when set) in the `ContentKeyDelegate` instead of the normal license acquisition.

I am new to Swift, so I may overlooked things. Please let me know if so. 🙂 

# Test Plan

There are no tests for the new `expo-video` library yet.

### CODE REVIEW

File: packages/expo-video/ios/Records/DRMOptions.swift [hunk 1]
```diff
@@ -10,6 +10,9 @@ internal struct DRMOptions: Record {
   @Field
   var licenseServer: String?
 
+  @Field
+  var getLicense: ((String, String, String) -> Promise<String>)?
```
Comment (tsapeta) [inline packages/expo-video/ios/Records/DRMOptions.swift:14]: Have you tested this? As far as I remember records don't support passing functions yet 😅

Comment (JustJoostNL) [reply packages/expo-video/ios/Records/DRMOptions.swift:14]: I am on Windows my self, so I couldn't find an easy way to test. But I guess I'll need to find another way for that function haha

Comment (JustJoostNL) [reply packages/expo-video/ios/Records/DRMOptions.swift]: @tsapeta I've migrated to an event now. Although, I am not sure if the Modules API supports using data returned from an event. 😅

**Review by tsapeta — COMMENTED** [review]
Hey @JustJoostNL, I apologize that we need to close this PR for now. Neither in RN modules nor Expo modules is possible for the view event to return something. It's basically a limitation of the old architecture.

I have another idea how we could approach this, but I would rather prefer if @behenate could take care of it instead. Especially that you don't have macOS, so you can't test it properly. I've already shared my idea with him.

Anyway, I really appreciate your willingness and commitment on this! 🙇‍♂️

**Comment by JustJoostNL** [comment]
Hi @tsapeta

All good! I understand that me not being on macOS and new to Swift isn't ideal for this. So it will be better indeed if Behenate takes care of this one. Though, would you mind sharing your approach idea with me? Just trying to learn here. 🙂

**Comment by tsapeta** [comment]
Sure! I'm not quite sure if this will work perfectly, but in theory we could use the `JavaScriptFunction<ReturnType>` class as an argument type in a synchronous function and keep it in the native player to call it later. Something like this:

```swift
// Swift
Function("setGetDRMLicense") { (player: VideoPlayer, callback: JavaScriptFunction<String>) in
  player.getDRMLicense = callback
}

// and then call it somewhere
let ckcString: String = player.getDRMLicense.call(params)
```

```js
// JS
player.setGetDRMLicense((params) => "ckcString");
```

**Comment by JustJoostNL** [comment]
Ah, thanks! Is there any ETA for this btw (it's a pretty important feature for us)? Though, I understand that of course other features may have higher priority at the moment.

**Pull request closed without merging**